app/src/main/java/com/arctouch/codechallenge/home/HomeAdapter.kt

```diff
@@ -21,6 +24,11 @@ class HomeAdapter(private val movies: List<Movie>) : RecyclerView.Adapter<HomeAd
             itemView.titleTextView.text = movie.title
             itemView.genresTextView.text = movie.genres?.joinToString(separator = ", ") { it.name }
             itemView.releaseDateTextView.text = movie.releaseDate
+            itemView.cardMovie.setOnClickListener{
```

Comment:
Imagine that tapping on the card could have different results depending on some business rule. How would you (unit) test such business rule and its effects on the view? Do you foresee any challenges with your current implementation?

Reply:
You are right, in this case the implementation was not scalable, the biggest challenge would be the large code increase to implement different business rules.